[refactor] #225: redesign snippets workflow (#232)
hyperledger/iroha-2-docs: src/documenting/snippets.md

[Code Snippets](https://vitepress.vuejs.org/guide/markdown#import-code-snippets)
feature:

**Input**

```md
<<<@/snippets/lorem.rs

<<<@/snippets/lorem.rs#ipsum
```

**Output**

<<<@/snippets/lorem.rs

<<<@/snippets/lorem.rs#ipsum

Pay attention to how we included only `#ipsum` code region. This feature is
essential when it comes to including real source files content into the
documentation.

## Example

Let's add a code snippets from the Iroha JavaScript SDK, for example
[`/packages/docs-recipes/src/1.client-install.ts`](https://github.com/hyperledger/iroha-javascript/blob/e300886e76c777776efad1e2f5cb245bfb8ed02e/packages/docs-recipes/src/1.client-install.ts).
To do so, the first thing we need is to get a permalink to the file.
Clicking by the "Raw" button at GitHub, we got it:

- https://raw.githubusercontent.com/hyperledger/iroha-javascript/iroha2/packages/docs-recipes/src/1.client-install.ts

Then, we define the new snippet in the [Snippet Sources](#snippet-sources):

```json
{
  "src": "https://raw.githubusercontent.com/hyperledger/iroha-javascript/iroha2/packages/docs-recipes/src/1.client-install.ts",
  "filename": "js-sdk-1-client-install.ts"
}
```

> Then, it can be [included](#using-snippets-in-markdown) in any Markdown
> file in the documentation as follows:

Reviewer: ```suggestion
3. [Include](#using-snippets-in-markdown) the snippet in any Markdown
file in the documentation as follows:
```

# Code Snippets

To make code snippets in the documentation more "real" and robust, it is
better to fetch them directly from the source files. The sources are
located in other repositories, where they are built, run, and tested.

## How it works

### Snippet Sources

Snippet sources are defined in
[`snippet_sources.json`](https://github.com/hyperledger/iroha-2-docs/blob/main/snippet_sources.json).
The `snippet_sources.json` file is located at the documentation repository
root and has the following format:

```json
[
  {
    "src": "https://raw.githubusercontent.com/hyperledger/iroha/iroha2-stable/MAINTAINERS.md",
    "filename": "iroha-maintainers-at-stable.md"
  },
  {
    "src": "./src/example_code/lorem.rs"
  }
]
```

- `src` defines the source file location and could be either an HTTP(s) URI
  or a relative file path.
- `filename` (optional) explicitly defines the local filename.

### Fetching Snippets

Code snippets are fetched from the locations specified in
`snippet_sources.json` and written into the `/src/snippets` directory.
There are two ways to fetch the snippets:

> - Automatically after packages installation (i.e. `pnpm install`)

Reviewer: ```suggestion
- Automatically after package installation (i.e. `pnpm install`)
```

Author: [`pnpm.io`](https://pnpm.io/cli/install) says:

> `pnpm install` is used to install all dependencies for a project.

So, maybe its better?

```suggestion
- Automatically after dependencies installation (i.e. `pnpm install`)
```

Reviewer: ```suggestion
- Automatically after dependencies were installed (i.e. `pnpm install`)
```
